File: CSharp/Nova/IO/NOVFile.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Nova.IO
{
    public class NOVFile
    {
        private string Filepath
        {
            get;
            set;
        }
        public NOVFile(string filepath)
        {
            this.Filepath = filepath;
            this.Deserialize();
        }

        private void Deserialize()
        {
            FileStream stream = new FileStream(Filepath, FileMode.Open);
            BinaryReader reader = new BinaryReader(stream);


            float test2 = reader.ReadSingle();
            stream.Close();
            reader.Close();
        }
    }
}

File: Virtual Machine/Nova.VM/NovFile.cpp
#include "NovFile.h"
#include "Logger.h"


NovFile::NovFile()
{

}

bool NovFile::Deserialize(BinaryReader& reader)
{
	std::string header = reader.ReadString();

	if (header != NovFile::HEADER)
	{
		Logger::Error("Invalid file. Header corrupted.");
		return false;
	}



	return true;
}
